pkgs/tools/security/sequoia/default.nix

```diff
@@ -0,0 +1,74 @@
+{ stdenv, fetchFromGitLab, git, nettle, llvmPackages_8, cargo, rustc
+, rustPlatform, pkgconfig, glib
+, openssl, sqlite, capnproto
+, pythonSupport ? true, python37Packages ? null
+}:
+
+assert pythonSupport -> python37Packages != null;
+
+rustPlatform.buildRustPackage rec {
+  version = "0.9.0";
+  pname = "sequoia";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.com";
+    owner = "sequoia-pgp";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "13dzwdzz33dy2lgnznsv8wqnw2501f2ggrkfwpqy5x6d1kgms8rj";
+  };
+  cargoSha256 = "1zcnkpzcar3a2fk2rn3i3nb70b59ds9fpfa44f15r3aaxajsdhdi";
+
+  pyenv = if pythonSupport then
```

Comment:
When generating bindings we should not use `python.withPackages` because the packages may be needed during run-time. Add the run-time deps to `propagatedBuildInputs` when `pythonSupport`

Reply:
and use lib.optionals